parameter_map_from should report which name is missing

### Description

I would like improved error messages for missing parameters. Currently I only see "No parameter value set". But I don't know which parameter this applies to. I'm trying to configure Nav2, so this is somewhat of an issue (...).

I tried to find my way around this code base:
In parameter_map.cpp the method parameter_map_from calls parameter_value_from.
That method can throw rclcpp::exceptions::InvalidParameterValueException. The description "No parameter value set" does not describe which parameter was invalid. That could be fixed: In parameter_map_from the c_param_name is known.

Does this sound like a feasible improvement?

### Motivation

I would like to improve debugging of missing parameters.

### Design / Implementation Considerations

_No response_

### Additional Information

_No response_

For example:
```
1753634977.9965954 [planner_server-11] terminate called after throwing an instance of 'rclcpp::exceptions::InvalidParameterValueException'
1753634978.0338519 [planner_server-11]   what():  No parameter value set
1753634980.9152861 [ERROR] [planner_server-11]: process has died [pid 24056, exit code -6, cmd '/opt/ros/jazzy/lib/nav2_planner/planner_server --ros-args -r __node:=planner_server -r __ns:=/ --params-file /home/pi/karaburan/ros_ws/src/install/navigation/share/navigation/config/planner_server.yaml'].
```
I would like to know which parameter has no value set. I don't want to be guessing what I'm not doing right. This is all the information I'm getting right now...

--- a/rclcpp/src/rclcpp/parameter_map.cpp
+++ b/rclcpp/src/rclcpp/parameter_map.cpp
@@ -89,7 +89,7 @@ rclcpp::parameter_map_from(const rcl_params_t * const c_params, const char * nod
           std::string("parameter_value_from failed for parameter '") +
           c_param_name + "': " + e.what());
       }
-      parameters.emplace_back(c_param_name, value);
+      params_node.emplace_back(c_param_name, value);
     }
   }
 

--- a/rclcpp/src/rclcpp/parameter_map.cpp
+++ b/rclcpp/src/rclcpp/parameter_map.cpp
@@ -84,8 +84,8 @@ rclcpp::parameter_map_from(const rcl_params_t * const c_params, const char * nod
       ParameterValue value;
       try {
         value = parameter_value_from(param);
-      } catch (const rclcpp::exceptions::InvalidParameterValueException & e) {
-        throw rclcpp::exceptions::InvalidParameterValueException(
+      } catch (const InvalidParameterValueException & e) {
+        throw InvalidParameterValueException(
           std::string("parameter_value_from failed for parameter '") +
           c_param_name + "': " + e.what());
       }

--- a/rclcpp/src/rclcpp/parameter_map.cpp
+++ b/rclcpp/src/rclcpp/parameter_map.cpp
@@ -83,7 +83,7 @@ rclcpp::parameter_map_from(const rcl_params_t * const c_params, const char * nod
       const rcl_variant_t * const c_param_value = &(c_params_node->parameter_values[p]);
       ParameterValue value;
       try {
-        value = parameter_value_from(param);
+        value = parameter_value_from(c_param_value);
       } catch (const InvalidParameterValueException & e) {
         throw InvalidParameterValueException(
           std::string("parameter_value_from failed for parameter '") +

--- a/rclcpp/src/rclcpp/parameter_map.cpp
+++ b/rclcpp/src/rclcpp/parameter_map.cpp
@@ -81,7 +81,7 @@ rclcpp::parameter_map_from(const rcl_params_t * const c_params, const char * nod
         throw InvalidParametersException(message);
       }
       const rcl_variant_t * const c_param_value = &(c_params_node->parameter_values[p]);
-      rclcpp::ParameterValue value;
+      ParameterValue value;
       try {
         value = parameter_value_from(param);
       } catch (const rclcpp::exceptions::InvalidParameterValueException & e) {

--- a/rclcpp/src/rclcpp/parameter_map.cpp
+++ b/rclcpp/src/rclcpp/parameter_map.cpp
@@ -81,7 +81,15 @@ rclcpp::parameter_map_from(const rcl_params_t * const c_params, const char * nod
         throw InvalidParametersException(message);
       }
       const rcl_variant_t * const c_param_value = &(c_params_node->parameter_values[p]);
-      params_node.emplace_back(c_param_name, parameter_value_from(c_param_value));
+      rclcpp::ParameterValue value;
+      try {
+        value = parameter_value_from(param);
+      } catch (const rclcpp::exceptions::InvalidParameterValueException & e) {
+        throw rclcpp::exceptions::InvalidParameterValueException(
+          std::string("parameter_value_from failed for parameter '") +
+          c_param_name + "': " + e.what());
+      }
+      parameters.emplace_back(c_param_name, value);
     }
   }
 

